Makefile

```
@@ -886,6 +886,18 @@ docker: DOCKER_MODULES ?= $(OCRD_MODULES)
 docker: DOCKER_PARALLEL ?= -j1
 docker: docker-latest
 
+OCRD_NETWORK_CONFIG ?= run-network/odem-workflow-config.yaml
+.PHONY: run-network
+run-network:
+	@$(PYTHON) -m venv $(VIRTUAL_ENV)
+	@$(VIRTUAL_ENV)/bin/pip install click requests pyaml shapely==1.8.5 ocrd
+	@$(VIRTUAL_ENV)/bin/python run-network/creator.py create-docker $(OCRD_NETWORK_CONFIG)
+	@$(VIRTUAL_ENV)/bin/python run-network/creator.py create-env $(OCRD_NETWORK_CONFIG)
+	@$(VIRTUAL_ENV)/bin/python run-network/creator.py start $(OCRD_NETWORK_CONFIG)
+
+	@$(VIRTUAL_ENV)/bin/python run-network/creator.py create-clients $(VIRTUAL_ENV)/bin $(OCRD_NETWORK_CONFIG)
```

I would recommend differentiating between setup phase (just once) and runtime (repeatable) here. Also, we should provide a teardown and a cleanup phase. Also, I would not mix this with our existing `VIRTUAL_ENV` variable (in contrast to which this time we fully control the venv).

How about something like this:
```suggestion
network-setup:
	$(PYTHON) -m venv run-network/.venv
	run-network/.venv/bin/python -m pip install click requests pyaml shapely==1.8.5 ocrd
	run-network/.venv/bin/python run-network/creator.py create-compose $(OCRD_NETWORK_CONFIG)
	run-network/.venv/bin/python run-network/creator.py create-dotenv $(OCRD_NETWORK_CONFIG)
	run-network/.venv/bin/python run-network/creator.py create-clients run-network/.venv/bin $(OCRD_NETWORK_CONFIG)

network-start:
	run-network/.venv/bin/python run-network/creator.py start $(OCRD_NETWORK_CONFIG)

network-stop:
	run-network/.venv/bin/python run-network/creator.py stop $(OCRD_NETWORK_CONFIG)

network-clean:
	$(RM) -r run-network/.venv run-network/.env run-network/docker-compose.yml

```

---

My suggestion advised splitting up the setup phase into file-based targets. The reason is simply that this allows selectively repeating/updating these steps. Also, if some file gets edited manually, the new time stamp will prevent it being overwritten. Also, adding more explicit dependencies (like on `OCRD_NETWORK_CONFIG`) could then further help manage this more smoothly.

You have not adopted this. Did my rules not work?

---

I think I missed the edit. Updated with: https://github.com/OCR-D/ocrd_all/pull/449/commits/b9d0f84a50a78325490e3ce136b1ff030b9692d2 and should fit now

---

Thanks!